lib/web_console/whiny_object.rb

```diff
@@ -0,0 +1,23 @@
+module WebConsole
+  class WhinyObject < SimpleDelegator
+    def initialize(object, env)
+      @env = env
+      super(object)
+    end
+
+    private
+
+      attr_accessor :env
+
+      def whine_unless(condition)
```

Comment:
If you buy this commit https://github.com/rails/web-console/commit/60eb494473a110af7bb6b235a7a84efb20d77b3a, I think this discussion is not relevant now.